### PR TITLE
change Mac simulatorApp from debug to release

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/cocos-engine/issues/11726

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by dumganhar** [comment]
Should CC_DEBUG macro be enabled in release mode only for Simulator?
Enable CC_DEBUG will get more log output.

Also https://github.com/cocos/cocos-engine/pull/11727 add a new cmake configuration called `USE_V8_DEBUGGER_FORCE `, need to `-DUSE_V8_DEBUGGER_FORCE=1` in the command line.

**Comment by PPpro** [comment]
This may change the output path of simulator executable program

Need @VisualSJ to update the reference path in Editor ?

**Comment by PPpro** [comment]
Oops, I've seen the change, maybe it's better to link the PR here https://github.com/cocos/cocos-editor/pull/426

File: native/gulpfile.js [hunk 1]
```diff
@@ -147,11 +147,10 @@ gulp.task('clean-simulator', async function () {
     let isWin32 = process.platform === 'win32';
     let delPatterns = [
         formatPath(Path.join(__dirname, './simulator/*')),
-        formatPath(`!${Path.join(__dirname, './simulator/Debug')}`),
+        formatPath(`!${Path.join(__dirname, './simulator/Release')}`),
     ];
     if (!isWin32) {
-        delPatterns.push(formatPath(Path.join(__dirname, './simulator/Debug/libcocos2d.a')));
```
Comment (PPpro) [inline native/gulpfile.js:153]: we don't generate libcocos2d.a any more ?

Comment (novain1102) [reply native/gulpfile.js:153]: Yes.

**Review by PPpro — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Comment by novain1102** [comment]
dont merge, still being modified

**Review by dumganhar — APPROVED** [review]

**Comment by novain1102** [comment]
> dont merge, still being modified

done.

**Review by PatriceJiang — APPROVED** [review]

**Pull request merged**